### PR TITLE
HIPP-1595 - Deploy API - Producer

### PR DESCRIPTION
**Pull request by victorarbuesmallada**

### CODE REVIEW

File: app/controllers/myapis/produce/ProduceApiCheckYourAnswersController.scala [hunk 1]
```diff
@@ -38,34 +45,142 @@ class ProduceApiCheckYourAnswersController @Inject()(
...
+        passthrough <- validatePassthrough(request.userAnswers, request.user.permissions.canSupport)
+        apiStatus <- validateApiStatus(request.userAnswers)
+        domainSubdomain <- validateDomainSubdomain(request.userAnswers)
+        hods <- validateHods(request.userAnswers)
+        egressPrefixes <- validateEgressPrefixes(request.userAnswers)
+    } yield DeploymentsRequest(
+      lineOfBusiness = "apim",
+      name = apiTitle,
+      description = shortDescription,
+      egress = egress.egress.getOrElse("unassigned"),
```
Comment (victorarbuesmallada) [inline app/controllers/myapis/produce/ProduceApiCheckYourAnswersController.scala:86]: I've seen that there is some logic around the `unassigned` value on the [egress form](https://github.com/hmrc/api-hub-frontend/blob/main/app/forms/myapis/produce/ProduceApiChooseEgressFormProvider.scala#L37-L48), not sure if I should change that logic

File: app/services/ApiHubService.scala [hunk 2]
```diff
@@ -218,4 +225,12 @@ class ApiHubService @Inject()(
     applicationsConnector.fetchAllScopes(applicationId)
   }
 
+  private[services] def formatAsKebabCase(text: String): String =
```
Comment (victorarbuesmallada) [inline app/services/ApiHubService.scala:228]: not sure if this should be here, on a utils class or if should be on the `api-hub-applications` side, happy to change it though

File: app/views/myapis/produce/ProduceApiCheckYourAnswersView.scala.html [hunk 3]
```diff
@@ -22,18 +23,38 @@
     formHelper: FormWithCSRF,
     govukButton: GovukButton,
     govukSummaryList: GovukSummaryList,
+    govukErrorSummary: GovukErrorSummary,
 )
 
-@(summaryList: SummaryList, userModel: UserModel)(implicit request: Request[?], messages: Messages)
+@(summaryList: SummaryList, userModel: UserModel, maybeFailuresResponse: Option[FailuresResponse] = None)(implicit request: Request[?], messages: Messages)
+
+@errorList(failuresResponse: FailuresResponse) = @{
```
Comment (victorarbuesmallada) [inline app/views/myapis/produce/ProduceApiCheckYourAnswersView.scala.html:31]: not sure if we want to display errors this way, it looks this way:
Screenshot from 2024-11-06 17-02-45

File: app/services/ApiHubService.scala (hunk 2 above)
Comment (PaulCDurham) [reply app/services/ApiHubService.scala:228]: Put this in the controller? That instantiates the DeploymentsRequest.

File: app/views/myapis/produce/ProduceApiCheckYourAnswersView.scala.html (hunk 3 above)
Comment (PaulCDurham) [reply app/views/myapis/produce/ProduceApiCheckYourAnswersView.scala.html:31]: Maybe pre tags around pretty printed JSON like we did when validating the OAS.

Comment (victorarbuesmallada) [reply app/views/myapis/produce/ProduceApiCheckYourAnswersView.scala.html]: Screenshot from 2024-11-07 15-36-56

**Pull request merged**